File: frontend/src/types/customConfig.ts
export interface CustomConfig {
  id: number;
  name: string;
  namespace: string;
  resourceType: string;
  yamlContent: string;
  description?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateCustomConfigPayload {
  name: string;
  namespace: string;
  resourceType: string;
  yamlContent: string;
  description?: string;
}

export interface UpdateCustomConfigPayload extends CreateCustomConfigPayload {}

File: backend/src/main/java/com/example/k8sconfig/entity/CustomConfig.java
package com.example.k8sconfig.entity;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class CustomConfig {
    private Long id;
    private String name;
    private String namespace;
    private String resourceType;
    private String yamlContent;
    private String description;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
